lms/djangoapps/courseware/module_render.py

```diff
@@ -388,7 +386,9 @@ def _fulfill_content_milestones(user, course_key, content_key):
             entrance_exam_enabled = getattr(course, 'entrance_exam_enabled', False)
             in_entrance_exam = getattr(content, 'in_entrance_exam', False)
             if entrance_exam_enabled and in_entrance_exam:
-                exam_pct = _calculate_entrance_exam_score(user, course)
+                request = RequestFactory()
```

Comment:
@martynjames -- yeah, I agree that this is not exactly optimal.  However, it seems we don't have access to the regular request object when in the context of the publish() workflow.  So, this approach allows us to accommodate the  downstream dependencies on request (see get_module_for_descriptor) introduced during the dynamic determination of exam modules used in scoring.  

I'm certainly open to better ways of doing this :)